# When a mirror causes timed out, cnchi was failing to continue
# trying with other mirrors.
# This patch needs testing.

diff --git a/src/pacman/pac.py b/src/pacman/pac.py
index 7b943b7..123a1a2 100644
--- a/src/pacman/pac.py
+++ b/src/pacman/pac.py
@@ -145,14 +145,12 @@ class Pac(object):
                     self.add_package(pkgname)
                 try:
                     self.t.prepare()
-                    self.t.commit()
-                    self.t.release()
+                    self.t.commit()                   
                     self.t_lock = False
                 except pyalpm.error:
-                    self.t.release()
-                    self.t_lock = False
                     line = traceback.format_exc()
                     self.queue_event("error", line)
+                self.t.release()
     
     def queue_event(self, event_type, event_text=""):
         self.callback_queue.put((event_type, event_text))
@@ -222,8 +220,6 @@ class Pac(object):
         if level & pyalpm.LOG_ERROR:
             self.error = _("ERROR: %s") % line
             print(line)
-            self.t.release()
-            self.t_lock = False
             self.queue_event("error", line)
         elif level & pyalpm.LOG_WARNING:
             self.warning = _("WARNING: %s") % line
